packages/store-ui/src/molecules/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,31 @@
+import type { ReactNode } from 'react'
+import React, { forwardRef } from 'react'
+
+import Button from '../../atoms/Button'
+import type { ButtonProps } from '../../atoms/Button'
+import Icon from '../../atoms/Icon'
+
+export interface Props extends ButtonProps {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+  /**
+   * The icon component
+   */
+  icon: ReactNode
+  children?: never
+}
```

Comment:
if you don't wanna has children, you just can use Omit
```suggestion
export interface Props extends Omit<ButtonProps, 'children'> {
  /**
   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
   */
  testId?: string
  /**
   * The icon component
   */
  icon: ReactNode
}
```

But I think we must keep the children, because in some cases we use Icon + text, as an add button...

Reply:
Every component receives children, is the default of React, to remove it on typescript I did this trick that explicitly removes the children props.

Reply:
I know, what I meant is: u can remove the children prop using Omit utility by typescript

So, actually, you use it like this:

```tsx
export interface Props extends ButtonProps {
  // ...props
  children?: never
}
```

And my suggestion is:

```tsx
export interface Props extends Omit<ButtonProps, 'children'> {
  // ...props
}

Reply:
Yeah, I understand that, but why do you prefer to use Omit being that uses never too? Maybe ain't seeing the advantage of using it. The omit will create a type with `children: never`, because of it I'm asking that.

```ts
type Exclude<T, U> = T extends U ? never : T;
type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
```

Reply:
But ok, I can change.